test/test_transformers.py

```diff
@@ -997,6 +1041,272 @@ def test_script_mha_in_proj_weight_none(self):
         torch.jit.script(mha)
 
 
+class TestSDPAFailureModes(NNTestCase):
+    """ Used to test the failure modes of scaled_dot_product_attention
+    """
+    _do_cuda_memory_leak_check = True
+    _do_cuda_non_default_stream = True
+
+    @unittest.skipIf(not PLATFORM_SUPPORTS_FUSED_SDPA or not isSM86or89Device,
+                     "Does not support fused SDPA or not SM86+ hardware")
+    @parametrize("head_dim", [72, 96, 128])
+    def test_memory_efficient_sm86_plus_failure(self, head_dim: int):
```

Comment:
Can we just have a `device` argument and decorate with `@onlyCUDA`?

Reply:
I did a largeish refactor to fix the TODO and use instantiate_device_type_tests. For this test I used @onlyCUDA. I don't need to do the name changes right? Since instantiate_device_type_tests will stub out the _cuda version test?